### PR TITLE
Fix Python client TCPTransports not being able to close

### PR DESCRIPTION
**Pull request by Thom747**
# Pull request

**Choose Correct**

- [X] bug
- [ ] feature

**Describe the pull request**
This PR fixes a problem with the Python implementation of `TCPTransport` that prevented it from closing its TCP connection if it was created as a client.

As the field `_serverSockEventStart` is only initialized if the `TCPTransport` is created as a server, calling `close` on a client `TCPTransport` results in an `AttributeError` before `socket.close` is called.
Thus, the connection stays alive.

**To Reproduce**
Create a client `TCPTransport`, then attempt to call `close()` on it.

**Expected behavior**
The `TCPTransport` should close the underlying socket and thus the TCP connection.

**Desktop (please complete the following information):**

- OS: Ubuntu 22.04
- eRPC Version: v1.11.0

**Steps you didn't forgot to do**

- [X] I checked if other PR isn't solving this issue.
- [X] I read Contribution details and did appropriate actions.
- [X] PR code is tested.
- [X] PR code is formatted.
- [X] Allow edits from maintainers pull request option is set (recommended).

### CODE REVIEW

**Comment by Hadatko** [comment]
Hi, thank you for your PR. I will take a on it soon.

File: erpc_python/erpc/transport.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 import struct
 import socket
 import threading
+from _socket import SHUT_RDWR
```
Comment (Hadatko) [inline erpc_python/erpc/transport.py:13]: Hi, why you don't use value from socket? Not sure where is from _socket

File: erpc_python/erpc/transport.py [hunk 2]
```diff
@@ -166,7 +168,9 @@ def _serve(self):
             self._serverSockEventStart.set()
 
     def close(self):
-        self._serverSockEventStart.clear()
+        if self._isServer:
+            self._serverSockEventStart.clear()
+        self._sock.shutdown(SHUT_RDWR)
```
Comment (Hadatko) [inline erpc_python/erpc/transport.py:173]: i would use directly socket.SHUT_RDWR
image

**Review by Hadatko — CHANGES_REQUESTED** [review]

File: erpc_python/erpc/transport.py (hunk 1 above)
Comment (Thom747) [reply erpc_python/erpc/transport.py]: Ah, I originally couldn't find this value in `socket`, so I used the private module. This has been fixed now.

**Review by Hadatko — APPROVED** [review]

**Review by MichalPrincNXP — APPROVED** [review]

**Comment by MichalPrincNXP** [comment]
Thank you the effort

**Pull request merged**